pkg/referrerstore/oras/authprovider/azure/azureworkloadidentity.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/confidential"
+	provider "github.com/deislabs/ratify/pkg/referrerstore/oras/authprovider"
+	"github.com/pkg/errors"
+)
+
+type AzureWIProviderFactory struct{}
+type azureWIAuthProvider struct{}
+
+type azureWIAuthProviderConf struct {
+	Name string `json:"name"`
+}
+
+const (
+	azureWIAuthProviderName      string = "azure-wi"
+	dockerTokenLoginUsernameGUID string = "00000000-0000-0000-0000-000000000000"
+	AADResource                  string = "https://containerregistry.azure.net"
+)
+
+// init calls Register for our Azure Workload Identity provider
+func init() {
+	provider.Register(azureWIAuthProviderName, &AzureWIProviderFactory{})
+}
+
+// Create returns an AzureWIAuthProvider
+func (s *AzureWIProviderFactory) Create(authProviderConfig provider.AuthProviderConfig) (provider.AuthProvider, error) {
+	conf := azureWIAuthProviderConf{}
+	authProviderConfigBytes, err := json.Marshal(authProviderConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := json.Unmarshal(authProviderConfigBytes, &conf); err != nil {
+		return nil, fmt.Errorf("failed to parse auth provider configuration: %v", err)
+	}
+
+	return &azureWIAuthProvider{}, nil
+}
+
+// Enabled always returns true since there are no fields to verify
+func (d *azureWIAuthProvider) Enabled() bool {
+	return true
+}
+
+// Provide returns the credentials for a specificed artifact.
+// Uses Azure Workload Identity to retrieve an AAD access token which can be
+// exchanged for a valid ACR refresh token for login.
+func (d *azureWIAuthProvider) Provide(artifact string) (provider.AuthConfig, error) {
+	tenantID := os.Getenv("AZURE_TENANT_ID")
```

Comment:
Do you need to validate if the value is not empty?

Reply:
Yes I'll add this validation to the enabled method